core/templates/dev/head/exploration_editor/main_tab/StateResponses.js

```diff
@@ -284,20 +284,22 @@ oppia.controller('StateResponses', [
       trainingDataService.initializeTrainingData(
         explorationId, currentStateName);
     };
+
     $scope.suppressDefaultAnswerGroupWarnings = function() {
       var interactionId = $scope.getCurrentInteractionId();
+      var answerGroups = responsesService.getAnswerGroups();
+      var handledAnswersArray = [];
+      var choiceIndices = [];
+      var answerChoices = [];
       if (interactionId === 'MultipleChoiceInput') {
-        var answerGroups = responsesService.getAnswerGroups();
+        var numChoices = $scope.getAnswerChoices().length;
```

Comment:
If this is the same in both branches, you can pull it out, like you did with choiceIndices etc.